File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int index = 1;
        int cnt = 1;

        for (int idx = 1; idx < nums.length; ++idx) {
            if (nums[idx] == nums[idx - 1]) {
                cnt++;
            } else {
                cnt = 1;
            }

            if (cnt == 1) {
                nums[index++] = nums[idx];
            }
        }

        System.out.println(Arrays.toString(nums));

        return index;
    }
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        
        int index = 0;
        int cnt = 0;
        int prev = nums[0];

        for (int i = 0; i < nums.size(); i++) {

            int cur = nums[i];

            if (prev == cur) cnt++;

            if (cnt == 1) {
                nums[index++] = cur;
            } else {
                cnt = 1;
            }

            prev = cur;
        }

        return index;
    }
};


/*

index = 0
cnt = 1
int prev = 0

[0,0,1,1,1,2,2,3,3,4]
    cur = nums[i]

    if (prev == cur) cnt++;

    if (cnt == 1) { 
        nums[index++] = cur;
    } else {
        cnt = 0
    }



**/